server/index.js

```diff
@@ -294,6 +294,32 @@ app.get('/api/posts', (req, res, next) => {
     .catch(err => next(err));
 });
 
+app.delete('/api/posts/:postId', (req, res, next) => {
+  const { userId } = req.user;
+  const postId = Number(req.params.postId);
+
+  if (!userId) {
+    throw new ClientError(400, 'could not find user');
+  }
```

Comment:
This won't hurt anything but shouldn't be necessary. Your authMiddleware already verifies the `user` info so there is no need to check it again in your route handlers.